### PR TITLE
Adding unit-tests

### PR DESCRIPTION
**Pull request by mbad0la**
**Work in Progress**

This PR is in reference to #5 

@paazmaya should I make provisions to automatically close the issue once this PR gets merged?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -14,7 +14,12 @@ const yaml = require('js-yaml');
 const check = require('check-type').init();
 
 const YamlValidatore = function YamlValidatore(options) {
-  this.options = options;
+  this.options = Object.assign({
```
Comment (mbad0la) [inline index.js]: Yeah! Creating tests do help to identify trouble spots in the code :)

I'm intending to put in a few more tests, then maybe you can merge this PR?

Comment (paazmaya) [inline index.js]: Just let me know when you consider it is ready

**Comment by mbad0la** [comment]
Hey @paazmaya 

I'm done with my additions. Please review and merge. Cheers!

File: tests/index_test.js [hunk 2]
```diff
@@ -9,12 +9,205 @@
 'use strict';
 
 const tape = require('tape'),
-  validator = require('../index');
+  validator = require('../index'),
+  fs = require('fs');
```
Comment (paazmaya) [inline tests/index_test.js]: Separate native modules from other modules by listing them in a separate const that appears as first

File: tests/index_test.js [hunk 3]
```diff
@@ -9,12 +9,205 @@
...
+
+  test.equal(typeof validatorInstance.errored, 'function');
+  test.equal(typeof validatorInstance.validateStructure, 'function');
+  test.equal(typeof validatorInstance.loadFile, 'function');
+  test.equal(typeof validatorInstance.checkFile, 'function');
+  test.equal(typeof validatorInstance.validate, 'function');
+  test.equal(typeof validatorInstance.report, 'function');
+});
+
+tape('Default paramters\' values', test => {
```
Comment (paazmaya) [inline tests/index_test.js]: parameter

File: tests/index_test.js [hunk 4]
```diff
@@ -9,12 +9,205 @@
...
+  test.plan(4);
+
+  const validatorInstance = new validator();
+  validatorInstance.validate([]);
+
+  test.equal(validatorInstance.inValidFilesCount, 0);
+  test.equal(validatorInstance.logs.length, 0);
+  test.equal(validatorInstance.nonValidPaths.length, 0);
+  test.equal(typeof validatorInstance.options, 'object');
```
Comment (paazmaya) [inline tests/index_test.js]: perhaps could add a test case that checks the options keys and their default values, possibly also checking that given options are used

File: tests/index_test.js [hunk 5]
```diff
@@ -9,12 +9,205 @@
...
+  test.plan(1);
+
+  const options = {
+    writeJson: true
+  }
+
+  const validatorInstance = new validator(options);
+  validatorInstance.validate(['appveyor.yml']);
+
+  fs.unlink('appveyor.json', function (err) {
```
Comment (mbad0la) [inline tests/index_test.js]: @paazmaya `fs.unlink` returns an error if the file (to be deleted) does not exist. Hence it performs the role of checking and if found, deleting the file.

I have added another commit making changes requested by you. Please have a look at them.

**Comment by paazmaya** [comment]
Thank you very much for the contribution!

**Pull request merged**